test/torch/nn/test_rnn.py

```diff
@@ -0,0 +1,182 @@
+
+    # model hyperparameters
+    batch_size = 4
+    input_size = 3
+    hidden_size = 5
+    seq_len = 6
+    num_layers = 2
+    bidirectional = True
+    directions = 2 if bidirectional else 1
+
+    x = (
+        th.randn(batch_size, seq_len, input_size)
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+    model = (
+        RNN(
+            input_size=input_size,
+            hidden_size=hidden_size,
+            num_layers=num_layers,
+            nonlinearity=nonlinearity,
+            batch_first=True,
+            bidirectional=bidirectional,
+        )
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+
+    out, h = model(x)
+    assert out.get().float_precision().shape == th.Size(
+        [batch_size, seq_len, directions * hidden_size]
+    )
+    assert h.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+
+    h0 = (
+        th.randn(batch_size, directions * num_layers, hidden_size)
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+    out, h = model(x, h0)
+    assert out.get().float_precision().shape == th.Size(
+        [batch_size, seq_len, directions * hidden_size]
+    )
+    assert h.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+
+
+def test_gru(hook, workers):
+    bob = workers["bob"]
+    alice = workers["alice"]
+    crypto_prov = sy.VirtualWorker(hook, id="crypto_prov")
+
+    # model hyperparameters
+    batch_size = 4
+    input_size = 3
+    hidden_size = 5
+    seq_len = 6
+    num_layers = 2
+    bidirectional = True
+    directions = 2 if bidirectional else 1
+
+    x = (
+        th.randn(batch_size, seq_len, input_size)
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+    model = (
+        GRU(
+            input_size=input_size,
+            hidden_size=hidden_size,
+            num_layers=num_layers,
+            batch_first=True,
+            bidirectional=bidirectional,
+        )
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+
+    out, h = model(x)
+    assert out.get().float_precision().shape == th.Size(
+        [batch_size, seq_len, directions * hidden_size]
+    )
+    assert h.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+
+    h0 = (
+        th.randn(batch_size, directions * num_layers, hidden_size)
+        .fix_precision()
+        .share(alice, bob, crypto_provider=crypto_prov)
+    )
+    out, h = model(x, h0)
+    assert out.get().float_precision().shape == th.Size(
+        [batch_size, seq_len, directions * hidden_size]
+    )
+    assert h.get().float_precision().shape == th.Size(
+        [batch_size, directions * num_layers, hidden_size]
+    )
+
+
+def test_lstm(hook, workers):
```

Comment:
same comment.

Reply:
I can do only one test for GRU and LSTM and add the pytest decorator, but it might be a little bit messy as I will have some `if`s to handle it, since we have two states (hidden and cell) with LSTM and only one with GRU